src/vs/workbench/contrib/terminalContrib/accessibility/browser/textAreaSyncAddon.ts

```diff
@@ -100,9 +101,10 @@ export class TextAreaSyncAddon extends Disposable implements ITerminalAddon {
 			this._logService.debug(`TextAreaSyncAddon#updateCommandAndCursor: no line`);
 			return;
 		}
-		if (!!currentCommand) {
-			this._currentCommand = commandLine.substring(currentCommand.commandStartX ?? 0);
-			this._cursorX = buffer.cursorX - (currentCommand.commandStartX ?? 0);
+		const startX = isMacintosh ? currentCommand.commandStartX : 0;
+		if (!!currentCommand && !!startX) {
+			this._currentCommand = commandLine.substring(startX);
+			this._cursorX = buffer.cursorX - startX;
 		} else {
```

Comment:
Actually, it would be better to encapsulate any workarounds inside `CommandDetectionCapability` using `_isWindowsPty`. We don't want Windows quirks coming outside `CommandDetectionCapability` as then they'll end up being peppered all over the place.

Something I'm a little confused about with this is why command started would not come through at all, since if it does at all `commandStartX` should be a number:

https://github.com/microsoft/vscode/blob/414606b665b564d5ea4cce008531c7b23555bdd0/src/vs/platform/terminal/common/capabilities/commandDetectionCapability.ts#L365

Could you look at the sequences and see in which order it comes through? It might be firing before command finished as I think that's the only place we clear `commandStartX`?

https://github.com/microsoft/vscode/blob/414606b665b564d5ea4cce008531c7b23555bdd0/src/vs/platform/terminal/common/capabilities/commandDetectionCapability.ts#L517